components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -158,7 +158,7 @@ object typing {
   //   We can avoid this case by changing this folding logic - see the comment there
   case object TypedNull extends TypingResult {
 
-    override val withoutValue: TypingResult = this
+    override val withoutValue: TypingResult = Unknown
```

Comment:
Maybe let's add a comment, that `Unknown` is our `Any` and null value can be a subclass of any class so we return it.

Reply:
Maybe also let's add a notice in migration guide that we changed `TypedNull.wihoutValue` behaviour. As we can see in the core code, someone could base on the current behaviour.